meter.go

```diff
@@ -108,31 +107,22 @@ func getCounter(m *Metrics, counterName string) Counter {
 
 // GetJSON filters counters by glob pattern and returns them as a json marshaled
 // map.
-func (m *Metrics) GetJSON(pattern string) ([]byte, error) {
-	return getJSON(m, pattern)
+func (m *Metrics) GetJSON(predicate func(string) bool) ([]byte, error) {
+	return getJSON(m, predicate)
 }
 
-func getJSON(m *Metrics, pattern string) ([]byte, error) {
-	g, err := glob.Compile(pattern)
-	if err != nil {
-		return nil, err
-	}
-
+func getJSON(m *Metrics, predicate func(string) bool) ([]byte, error) {
 	result := make(map[string]int64)
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
 	for k, v := range m.counters {
-		if g.Match(k) {
+		if predicate(k) {
 			result[k] = v.Get()
 		}
 	}
 
-	b, err := json.Marshal(result)
-	if err != nil {
-		return nil, err
-	}
-	return b, nil
+	return json.Marshal(result)
```

Comment:
Instead of `json.Marshal` I'd suggest to use:

```go
package gometer

import (
	"bytes"
	"fmt"
)

type jsonFormatter struct{}

func (j *jsonFormatter) Format(counters map[string]Counter) []byte {
	var buf bytes.Buffer
	countersNum := len(counters)

	panicIfErr(buf.WriteByte('{'))

	for i, n := range sortedKeys(counters) {
		_, err := fmt.Fprintf(&buf, `"%v":%v`, n, counters[n].Get())
		panicIfErr(err)

		if i != countersNum-1 {
			panicIfErr(buf.WriteByte(','))
		}
	}

	panicIfErr(buf.WriteByte('}'))

	return buf.Bytes()
}

func panicIfErr(err error) {
	if err != nil {
		panic(err)
	}
}
```

`sortedKeys` can be found in `formatter.go`

As a result we'll have sorted metrics and reduce and number of allocation, that sounds good if there are tons of metrics.